daemon/algod/api/algod.oas2.json

```diff
@@ -1150,6 +1150,65 @@
         }
       }
     },
+    "/v2/transactions/simulate": {
+      "post": {
+        "consumes": [
+          "application/x-binary"
+        ],
+        "produces": [
+          "application/msgpack"
+        ],
+        "schemes": [
+          "http"
+        ],
+        "summary": "Simulates a raw transaction as it would be evaluated on the network.",
+        "operationId": "SimulateTransaction",
+        "parameters": [
```

Comment:
Shall we add a SimulationOptions struct now, even if it's nearly empty?  We'll eventually need a place for parameters like "run me in 'access any resource mode`" or "Run me against the latest block OR run me against the end of the current pending queue".

We'll also need a place for incoming state, so this becomes a suitable replacement for dry-run, but I suspect that ought to be a third parameter, to keep "options" separate from "state".

I am also ok with not doing this now.

Reply:
There's a problem if we want to have additional arguments in the request body (and if the goal is to specify initial state, that seems like it would need to be in the body, not on the URL).

Right now we mimic the raw transaction submission endpoint and treat the entire body as sign transaction bytes. There's no place for other arguments to be encoded.

We probably need to think of a way to also accept other arguments without differentiating too much from raw transaction endpoint.

Reply:
To be clear I don't think a change is necessary at the moment, but we should think about this and try to address it before we make the endpoint generally available

Reply:
I agree, we need some sort of "envelope" to contain the transaction(s) and metadata.  I find myself wanting this in the protocol itself often as well.